spec/src/main/asciidoc/ecosystem.asciidoc

```diff
@@ -0,0 +1,112 @@
+
+The properties defined in Kubernetes config map can be transformed to environment variables via envFrom. After a cloud-native microservice is deployed, if an extra config source is needed, this file can be mounted to configmap, which then use envFrom facility to expose the properties to the microservie. 
+
+[source, text]
+----
+kind: ConfigMap
+apiVersion: v1
+metadata:
+  name: example-config
+  namespace: default
+data:
+  example.property.1: hello
+  example.property.2: world
+  
+   
+----
+----
+# Use envFrom to load ConfigMaps into environment variables
+
+apiVersion: apps/v1beta2
+kind: Deployment
+metadata:
+  name: mans-not-hot
+  labels:
+    app: mans-not-hot
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: mans-not-hot
+  template:
+    metadata:
+      labels:
+        app: mans-not-hot
+    spec:
+      containers:
+        - name: app
+          image: gcr.io/mans-not-hot/app:bed1f9d4
+          imagePullPolicy: Always
+          ports:
+            - containerPort: 80
+          envFrom:
+          - configMapRef:
+              name: example-config
+----
+
+Therefore, config properties specified in the config map are automatically injectable to the microservices via MicroProfile Config APIs.
+
+= MicroProfile Health Check and Service Mesh
+In Service Mesh architecture, each pod has a lifecycle. Service Mesh needs to know when to kill a pod and when to route requests to a pod. Therefore, it needs to know each pod's health status. A pod's health status is measured using Liveness and Readiness.
+
+== Liveness
+Many microservices run for long periods of time and they eventually might transition to broken states, and cannot recover except being restarted. This is called liveness lifecycle.
+
+== Readiness
+
+Sometimes, microservices are temporarily unable to serve traffic. For example, an application might need to load large data or configuration files during startup. 
+
+== MicroProfile Health Check
+
+MicroProfile Health Check (https://github.com/eclipse/microprofile-health/) denotes whether the microservice is live or ready. It exposes an endpoint `/health`. Invoking the endpoint returns either UP (healthy) or DOWN (unhealthy).
+
+
+== Service Mesh usage of Liveness and Readiness
+
+Service Mesh e.g. Istio can utilise the readiness and liveness status from the underline component such as Kubernetes. 
+Kubernetes provides liveness probes or readiness probes to detect and remedy such situations. Kubernetes can check the pods frequently. 
+If the pod is not live, it will destroy the pod and start a new one. If the application is not ready, Kubernetes doesn’t want to kill the application, but not to send it requests either. 
+
+
+== The ecosystem
+Service mesh can utilise the endpoint exposed by MicroProfile Health for its liveness probe, so that it can control whether to destroy the pod or not.
```

Comment:
My understanding was that Service Mesh, in particular Istio, doesn't have it's own liveness probe.

The service liveness is still determined by Kubernetes liveness probe. Being a sidecar Istio is not involved in that at all